control/registration.go

```diff
@@ -0,0 +1,287 @@
+// Copyright 2025 Anapaya Systems
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package control
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/scionproto/scion/control/beacon"
+	"github.com/scionproto/scion/control/beaconing"
+	"github.com/scionproto/scion/pkg/addr"
+	"github.com/scionproto/scion/pkg/experimental/hiddenpath"
+	"github.com/scionproto/scion/pkg/log"
+	"github.com/scionproto/scion/pkg/metrics"
+	seg "github.com/scionproto/scion/pkg/segment"
+	"github.com/scionproto/scion/pkg/snet/addrutil"
+	"github.com/scionproto/scion/private/segment/seghandler"
+
+	"github.com/scionproto/scion/pkg/private/serrors"
+)
+
+type SegmentRegistrationPlugin interface {
+	// ID returns the unique identifier of the plugin.
+	ID() string
+	// New creates a new instance of the plugin with the provided configuration.
+	New(
+		ctx context.Context,
+		t *TasksConfig,
+		segmentType seg.Type,
```

Comment:
IMO we should not need those extra parameters. They should be part of the configuration.